lib/elixir/test/elixir/map_test.exs

```diff
@@ -113,6 +113,16 @@ defmodule MapTest do
     end
   end
 
+  test "merge with function" do
+    #when source map is bigger
+    assert Map.merge(%{a: 1, b: 2, c: 3}, %{c: 4, d: 5}, fn :c, 3, 4 -> :x end) ==
+             %{a: 1, b: 2, c: :x, d: 5}
+
+    # When target map is bigger
```

Comment:
I assume the `target` is the first argument as it'll get all keys from the second argument?